app/views/posts/show.html.erb

```diff
@@ -0,0 +1,16 @@
+<h2><%= link_to "Back to all posts", posts_path %></h2>
+
+<h1><%= @post.title %></h1>
+
+<p><%= @post.body %></p>
+<p><%= link_to "Edit Post", edit_post_path(@post) %></p>
+<p><%= link_to "Delete Post", post_path(@post), method: :delete %></p>
+
+<p><%= link_to "Add new comment", new_post_comment_path(@post) %></p>
+
+
+<% @post.comments.each do |comment| %>
+<p><%= comment.body %></p>
+<p><%= link_to "Edit comment", edit_post_comment_path(@post) %></p>
```

Comment:
The error you described is coming from Line 14. Rails is reading the post's id as the comments id (imo this behavior is dumb, it should throw an error* -- scroll to the end for my explanation of this thought). You need to pass in one more argument, the comment, so that the helper method `edit_post_comment_path` 'knows' what comment to edit. 

Fix: `edit_post_comment_path(@post, comment)`

*Since that path helper method takes two arguments, I think it should throw an error when it gets only 1.